RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -808,6 +849,10 @@ void MuonIdProducer::fillMuonId(edm::Event& iEvent, const edm::EventSetup& iSetu
    for ( const auto& chamber : info.chambers )
    {
      if (chamber.id.subdetId() == 3 && rpcHitHandle_.isValid()  ) continue; // Skip RPC chambers, they are taken care of below)
+
+     if (doME0_){
```

Comment:
a separate parameter in a separate place controls what goes into info.chambers.
It's better to not assume everything is set correctly and just skip the me0segments here completely.
Currently if the MuonIdProducer doMe0 is false, but if the associator doME0 is true, you will end up with ME0 chambers parsed in this block (and also in the specific ME0 block as well).